### PR TITLE
Improve CLI of Jucer2CMake (follow-up)

### PR DESCRIPTION
**Pull request by McMartin**
Improve certain things that were missed in https://github.com/McMartin/FRUT/pull/658.

@MartyLake: please review, thanks!

### CODE REVIEW

File: Jucer2CMake/src/main.cpp [hunk 1]
```diff
@@ -428,7 +428,13 @@ Arguments parseArguments(const int argc, const char* const argv[])
...
   {
     const auto& param = std::get<0>(paramAndValue);
-    if (!knownParams.contains(juce::String{param}))
+    if (knownFlags.contains(juce::String{param}))
+    {
+      const auto& value = std::get<1>(paramAndValue);
+      printError("unexpected argument \"" + value + "\" for \"" + param + "\"");
+      errorInArguments = true;
+    }
+    else if (!knownParams.contains(juce::String{param}))
```
Comment (MartyLake) [inline Jucer2CMake/src/main.cpp:437]: isnt this the exact else branche of l431?

Comment (MartyLake) [reply Jucer2CMake/src/main.cpp:437]: oh no one is flag and other is params

**Review by MartyLake — APPROVED** [review]
lgtm :)

**Comment by McMartin** [comment]
This PR is now based on top of https://github.com/McMartin/FRUT/pull/660.

**Comment by McMartin** [comment]
#660 has been merged and I have rebased this PR.

**Pull request merged**